pkg/testutil/TempDir.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 The Kubernetes Authors.
```

Comment:
1. file name is TempDir.go (should be all lowercase)
2. since this file is only referred by in the tests, I would add a `//+test` build tag so that this is not accessible from non-test builds. (I don't think calling it tempdir_test.go will solve it since it's cross-pkg, but I might be wrong).

Reply:
1. done
2. AFAIK the `test` build tag does not exist. I also searched and could not find anything. The only restriction could be to make it an internal package, but the import rules are too restrictive then. Also, `tempdir_test.go` is not possible. I'll leave as is for now.

Reply:
You're right, I guess it never happened.
To the least, maybe we can use `internal/testutil` so that it can't be imported  from outside?

Reply:
Doesn't work either. That will only allow imports exactly from the package where `internal` is in, so that would be `sigs.k8s.io/krew/pkg`, which is not useful.

I fully understand why you don't like it, but the helper cannot be properly hidden.